File: backend/Models/Domain/Article.cs
using System;
using System.Collections.Generic;

namespace _3w1m.Models.Domain;

public class Article
{
    public int ArticleId { get; set; } 
    public int CourseId { get; set; }
    public string Title { get; set; }
    public string? Summary { get; set; }
    public string Content { get; set; }
    
    public int Order { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    
    public Course Course { get; set; }
    public ICollection<ArticleProgress> ArticleProgresses { get; set; }
}

File: frontend/src/types/article.ts
export interface ArticleResponse {
  IsRead: boolean;
  ArticleId: number;
  CourseId: number;
  Title: string;
  Summary: string;
  Content: string;
  Order: number;
  CreatedAt: string;
  UpdatedAt: string;
  ReadAt: Date | null;
  CourseName: string;
}

export interface Article {
  id: string;
  courseId: string;
  title: string;
  summary: string;
  content: string;
  order: number;
  createdAt: string;
  updatedAt: string;
  isRead: boolean;
  readAt: Date | null;
  courseName: string;
}

export interface ArticleRequest {
  title: string;
  summary: string;
  content: string;
}
